Improbable/Stdlib/Improbable.Stdlib/WorkerConnection.cs

```diff
@@ -178,61 +174,58 @@ public void StopSendingMetrics()
 
         private static string GetDevelopmentPlayerIdentityToken(string host, ushort port, bool useInsecureConnection, string authToken, string playerId, string displayName)
         {
-            using (var pit = DevelopmentAuthentication.CreateDevelopmentPlayerIdentityTokenAsync(
+            using var pit = DevelopmentAuthentication.CreateDevelopmentPlayerIdentityTokenAsync(
                 host, port,
                 new PlayerIdentityTokenRequest
                 {
                     DevelopmentAuthenticationToken = authToken,
                     PlayerId = playerId,
                     DisplayName = displayName,
                     UseInsecureConnection = useInsecureConnection
-                }))
-            {
-                var value = pit.Get();
-
-                if (!value.HasValue)
-                {
-                    throw new AuthenticationException("Error received while retrieving a Player Identity Token: null result");
-                }
+                });
+            var value = pit.Get();
 
-                if (value.Value.Status.Code != ConnectionStatusCode.Success)
-                {
-                    throw new AuthenticationException($"Error received while retrieving a Player Identity Token: {value.Value.Status.Detail}");
-                }
+            if (!value.HasValue)
+            {
+                throw new AuthenticationException("Error received while retrieving a Player Identity Token: null result");
+            }
 
-                return value.Value.PlayerIdentityToken;
+            if (value.Value.Status.Code != ConnectionStatusCode.Success)
+            {
+                throw new AuthenticationException($"Error received while retrieving a Player Identity Token: {value.Value.Status.Detail}");
             }
+
+            return value.Value.PlayerIdentityToken;
         }
 
-        private static List<LoginTokenDetails> GetDevelopmentLoginTokens(string host, ushort port, bool useInsecureConnection, string workerType, string pit)
+        private static IEnumerable<LoginTokenDetails> GetDevelopmentLoginTokens(string host, ushort port, bool useInsecureConnection, string workerType, string pit)
```

Comment:
Might be a personal preference, but I would always return the highest type here (List<> in this case).
If a for each is done over the result it will use the List.Enumerator, while with the IEnumerable it wont.
Just a nit :P

Reply:
This is a interesting one. [Microsoft's guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections) suggest _not_ returning `List<T>`.

Rather they seem to favor using [`Collection<T>`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.collection-1?view=netframework-4.8) for [a couple of reasons](https://blogs.msdn.microsoft.com/codeanalysis/2006/04/27/faq-why-does-donotexposegenericlists-recommend-that-i-expose-collectiont-instead-of-listt-david-kean/). Namely that:

* `Collection<T>` has support for extensibility
* `List<T>` exposes properties/methods that wouldn't typically be applicable. 

I don't think the first reason isn't a concern for this. And the second seems iffy to me as well (optionality is a bad thing?).

I think I agree with @zeroZshadow on this particular method, since we actually receive the underlying data as a `List<>` and `IEnumerable<>` forces you to allocate if you ever want to do something _other than_ forward-iterate over the return value. 

I can't think of a benefit to returning an `IEnumerable<>` in this case (there's no deferred execution benefits that can come with `IEnumerable`) since its trivial for the caller to convert the `List` into the `IEnumerable` if they want.

Reply:
Given the intended use case of this method, I don't think allocations are really going to be an overriding decision factor. I'm going to stick with `IEnumerable<>`